src/SparkConduit.sol

```diff
@@ -160,9 +163,8 @@ contract SparkConduit is UpgradeableProxied, ISparkConduit, IInterestRateDataSou
     {
         require(getAvailableLiquidity(asset) == 0, "SparkConduit/non-zero-liquidity");
 
-        uint256 sharesToRequest = _convertToShares(asset, amount);
-
-        require(sharesToRequest <= shares[asset][ilk], "SparkConduit/amount-too-large");
+        // Limit remaining requested funds to the amount of shares
+        uint256 sharesToRequest = _min(shares[asset][ilk], _convertToShares(asset, amount));
```

Comment:
This is a change in functionality. Why is it being included in this PR? I don't necessarily disagree with the change, but this PR should be just the round up fix.

Also, this should be using the round-up version of `_convertToShares` as well.

Reply:
Made changes as discussed on call